Get build working again

Changes I need to be able to build successfully were backed out from yesterday to today.  Right now I get this build failure as a result of `make all`:
```
...
make -f cli/Makefile cliall
make[1]: Entering directory '/home/cribl/appscope'
DIR=/home/cribl/appscope/cli/
GO_VER=go1.15.6
GOPATH=/home/cribl/go
go get github.com/go-bindata/go-bindata/...
go get github.com/ahmetb/govvv
cp /home/cribl/appscope/cli/../bin/linux/scopec /home/cribl/appscope/cli/build/
cd /home/cribl/appscope/cli/ && go-bindata -o run/scopec.go -pkg run build/scopec
/bin/sh: 1: go-bindata: not found
cli/Makefile:36: recipe for target 'run/scopec.go' failed
make[1]: *** [run/scopec.go] Error 127
make[1]: Leaving directory '/home/cribl/appscope'
Makefile:29: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/cli/Makefile
+++ b/cli/Makefile
@@ -1,14 +1,13 @@
-GOVVV = $(shell command -v govvv 2>/dev/null)
-ifndef GOVVV
-    GO=go
-else 
-    GO=govvv
-endif
 
 DIR := $(dir $(realpath $(lastword $(MAKEFILE_LIST))))
 
 GO_VER:=$(shell go version | cut -d' ' -f3)
+
+# Allow GOPATH to be set, but don't require it.
+ifndef GOPATH
 GOPATH:=$(shell go env | grep GOPATH | cut -d\" -f2)
+endif
+
 GO:=go
 
 GO_VER_FLAGS=$(shell $(GOPATH)/bin/govvv -flags)
@@ -34,18 +33,18 @@ local:
 
 run/scopec.go:
 	cp $(DIR)../bin/linux/scopec $(DIR)build/
-	cd $(DIR) && go-bindata -o run/scopec.go -pkg run build/scopec
+	cd $(DIR) && $(GOPATH)/bin/go-bindata -o run/scopec.go -pkg run build/scopec
 
 scope: run/scopec.go
 	# GOOS=linux CGO_ENABLED=0 $(GO) build -a -tags netgo -ldflags="-s -w" -o build/scope
-	cd $(DIR) && GOOS=linux $(GO) build -o $(DIR)build/scope 
+	cd $(DIR) && GOOS=linux $(GO) build -ldflags="$(GO_VER_FLAGS)" -o $(DIR)build/scope
 	upx -9 $(DIR)build/scope
 
 install: 
 	cp $(DIR)build/scope /usr/local/bin/scope
 
 clitest:
-	cd $(DIR) && go test -v -coverprofile coverage.out ./...
+	cd $(DIR) && $(GO) test -v -coverprofile coverage.out ./...
 
 cover:
-	cd $(DIR) && go tool cover -html=coverage.out -o coverage.html
+	cd $(DIR) && $(GO) tool cover -html=coverage.out -o coverage.html

--- a/install_build_tools.sh
+++ b/install_build_tools.sh
@@ -618,6 +618,49 @@ apt_lcov_install() {
     fi
 }
 
+apt_go_exists() {
+    if go version &>/dev/null; then
+        echo "go is already installed; doing nothing for go."
+    else
+        echo "go is not already installed."
+        return 1
+    fi
+}
+
+apt_go_install() {
+    echo "Installing go."
+    sudo apt-get install -y software-properties-common
+    sudo add-apt-repository -y ppa:longsleep/golang-backports
+    sudo apt update
+    sudo apt-get install -y golang-go
+    if [ $? = 0 ]; then
+        echo "Installation of go successful."
+    else
+        echo "Installation of go failed."
+        FAILED=1
+    fi
+}
+
+apt_upx_exists() {
+    if upx --version &>/dev/null; then
+        echo "upx is already installed; doing nothing for upx."
+    else
+        echo "upx is not already installed."
+        return 1
+    fi
+}
+
+apt_upx_install() {
+    echo "Installing upx."
+    sudo apt-get install -y upx
+    if [ $? = 0 ]; then
+        echo "Installation of upx successful."
+    else
+        echo "Installation of upx failed."
+        FAILED=1
+    fi
+}
+
 apt_dump_versions() {
     # The crazy sed stuff at the end of each just provides indention.
     if lsb_release -d &>/dev/null; then
@@ -632,6 +675,8 @@ apt_dump_versions() {
     libtool --version | head -n1 | sed 's/^/      /'
     cmake --version | head -n1 | sed 's/^/      /'
     lcov --version | head -n1 | sed 's/^/      /'
+    go version | head -n1 | sed 's/^/      /'
+    upx --version | head -n1 | sed 's/^/      /'
 }
 
 apt_install() {
@@ -655,6 +700,13 @@ apt_install() {
     if ! apt_lcov_exists; then
         apt_lcov_install
     fi
+    if ! apt_go_exists; then
+        apt_go_install
+    fi
+    if ! apt_upx_exists; then
+        apt_upx_install
+    fi
+
 
     if (( FAILED )); then
         echo "Installation failure detected."

